contracts/interfaces/IAxelarAmplifierGateway.sol

```diff
@@ -14,38 +14,47 @@ interface IAxelarAmplifierGateway is IAxelarGMPGateway {
     \*********/
 
     enum CommandType {
-        ApproveContractCall,
+        VerifyMessages,
         RotateSigners
     }
 
-    struct ContractCallApprovalParams {
+    struct Message {
+        string messageId;
         string sourceChain;
         string sourceAddress;
         address contractAddress;
         bytes32 payloadHash;
     }
 
-    struct RotateSignersParams {
-        bytes32 nonce;
-        bytes newSigners;
+    struct Proof {
+        // Amplifier domain separator
+        // keccak256(chain_name || amplifier_router_address || axelar_chain_id)
+        bytes32 domainSeparator;
+        // Commitment to the signer set
+        bytes32 signerCommitment;
```

Comment:
involves a larger refactor but would be cleaner to move this into the auth module's proof type instead

Reply:
+1